### PR TITLE
(#160) Standardize on clang-format rules. Apply them to Certifier code base.

### PR DESCRIPTION
**Pull request by gapisback**
Starting from a near-empty `.clang-format` file,  containing just `Language:  Cpp` and `BasedOnStyle:  Google`, this change set brings-in various clang-format'ting rules, to standardize the coding style, indentation and formatting for the  Certifier code base. Most of the basic Google-style (such as use of 2-spaces for indentations v/s tabs etc.) have been retained. Some changes are to do with readability, avoiding single-line function calls / loops, and alignment of parameters in function declarations, definitions and function calls. We have standardized on clang-format-11 tool, which is available on both Linux and macOSX. Currently, code in third_party/ sub-dir are not being processed.

The main files to note are: .clang-format, added to top-level code-directory. `CI/scripts/check-srcfmt.sh` is a stand-alone script that uses the clang-format tool to verify that all files in the code base conform to the style(s) chosen. 'fixall' argument can be used to fix the formatting in any active files during development. 

This script is run at the start of the CI build-job and will fail the job if any formatting errors are found.

-----
**NOTE: To the reviewers**:

This change-set has been laid down in many individual commits showing the diffs introduced by the addition of just-one clang-formatting directive / rule. This way, it's clear exactly how the formatting changed for a specific rule.

Rather than looking at the full set of files-change, the better way to understand these changes is to walk the list of commits. And, to go through quickly salient diffs introduced by a specific commit.

The specific commits which may be 'controversial', eliciting discussions from code authors are these:

1. [Diffs from AlwaysBreakAfterReturnType: TopLevel - Most controversial …](https://github.com/vmware-research/certifier-framework-for-confidential-computing/pull/163/commits/c124a25bab46dbeddb482069efb0ab0ea961ec93)
2. [Diffs due to BreakBeforeBraces: Custom, which enables new brace rules…](https://github.com/vmware-research/certifier-framework-for-confidential-computing/pull/163/commits/fb963737b010a0db7a8f3f32a961adc7c0edbef4)
4. [Diffs from DerivePointerAlignment: false. PointerAlignment: right app…](https://github.com/vmware-research/certifier-framework-for-confidential-computing/pull/163/commits/54a7613a11b4a12747c1cd3efc36aeec3bcf02b5)

Many clang-formatting directives were essentially no-ops; i.e. did not produce any diffs in our code base. This means that our code base was already 'conforming' to that style. I still added those directives as a way to future-proof code formatting to continue to conform to that style.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+InsertTrailingCommas: None
+IndentWidth:     2
+
+KeepEmptyLinesAtTheStartOfBlocks: true
+
+MaxEmptyLinesToKeep: 2
+
+NamespaceIndentation: None
+
+PointerAlignment: Right
```
[the remark this answers is not in the document]
Comment (gapisback) [reply .clang-format:83]: Changing from `PointerAlignment: Right` to  `PointerAlignment: Left`

changes formatting, e.g. in `measure_binary()` function

from:

```
178   byte *file_contents = (byte *)malloc(size);
192   m->assign((char *)digest, (int)len);
```

to:

```
178   byte* file_contents = (byte*)malloc(size);
192   m->assign((char*)digest, (int)len);
```

[the remark this answers is not in the document]
Comment (gapisback) [reply .clang-format:83]: Recommendation: Change to `PointerAlignment: Left`

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+# Do not allow file-specific variations in pointer alignment.
+# Apply PointerAlignment consistently.
+DerivePointerAlignment: false
+
+FixNamespaceComments: true
+
+IndentCaseLabels: true
+IndentCaseBlocks: false
+IndentGotoLabels: false
```
Comment (gapisback) [inline .clang-format:71]: @jlmucb : "_IndentGotoLabels: false is fine_"

File: .clang-format [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+IndentPPDirectives: AfterHash
```
Comment (gapisback) [inline .clang-format:72]: @jlmucb : "I don't like "#    include <stdbool.h>"   again, wasted space for no readability benefit."

Comment (gapisback) [reply .clang-format:72]: Changing this from `IndentPPDirectives: AfterHash` to `IndentPPDirectives: None` (default), changes the output

from:

```
 41 #ifdef OE_CERTIFIER
 42 #  include "openenclave/attestation.h"
 43 #  include "openenclave/sealing.h"
 44 #endif
```

to:

```
 41 #ifdef OE_CERTIFIER
 42 #include "openenclave/attestation.h"
 43 #include "openenclave/sealing.h"
 44 #endif
```

Comment (gapisback) [reply .clang-format:72]: Recommendation: Go with John's preference: `IndentPPDirectives: None`

**Review by gapisback — COMMENTED** [review]
Summary note from @jlmucb :

If I didn't mention a rule I don't have a stong opinion either way or I missed it.  I feel most strongly about the ones I indicated in my original mail.

File: .clang-format [hunk 4]
```diff
@@ -0,0 +1,110 @@
+# #############################################################################
+# clang-format file for Certifier Framework code base
+# #############################################################################
+Language:      Cpp
+BasedOnStyle:  Google
+
+AllowAllArgumentsOnNextLine: false
+AllowAllParametersOfDeclarationOnNextLine: false
```
[the remark this answers is not in the document]
Comment (gapisback) [reply .clang-format:8]: John -- Unclear, what this 'large spacing' issue you remark here. Maybe we can look at some output formatting that you'd like to see changed.

**Comment by gapisback** [comment]
Final resolutions based on meeting between @jlmucb and @yelvmw and @gapisback :

1. AlignConsecutiveAssignments: false
2. AlignConsecutiveDeclarations: true
3. AfterControlStatement: Never 
4. AlignConsecutiveMacros: true
5. AllowAllParametersOfDeclarationOnNextLine: false
6. AlwaysBreakAfterReturnType: None
7. BinPackArguments: false
8. BinPackParameters: false
9. BreakBeforeBraces: Custom
10. BreakBeforeBraces: Attach
11. BreakBeforeBinaryOperators: NonAssignment
12. AllowShortFunctionsOnASingleLine: Inline
13. DerivePointerAlignment: false
14. PointerAlignment: Right
15. IndentGotoLabels: false
16. IndentPPDirectives: AfterHash
17.

**Review by yelvmw — APPROVED** [review]
LGTM! Thank you!

**Review by jlmucb — APPROVED** [review]

File: .clang-format (hunk 3 above)
Comment (gapisback) [reply .clang-format:72]: Update: During final review, John agreed that it's ok to have nested `#include`'s and `#define`'s indented, for readability. Outermost #define and #includes will not have extra spaces in them.

**Comment by gapisback** [comment]
The line-item `clang-format` rules have been consolidated and applied to `/main` HEAD. Integrated to `/main` unde PR #172 .

No further work is needed on this PR. However, we should retain this as a reference of the discussions and review comments / feedback, which lead to the final set of clang-formatting rules we settled on.

**Pull request closed without merging**